src/core/audio/AudioSdl.cpp

```diff
@@ -283,24 +304,98 @@ void AudioSdl::sdlInputAudioCallback(Uint8 *_buf, int _len) {
 
 #endif
 
+QString AudioSdl::setupWidget::s_systemDefaultDevice = QObject::tr("[System Default]");
+
 AudioSdl::setupWidget::setupWidget( QWidget * _parent ) :
 	AudioDeviceSetupWidget( AudioSdl::name(), _parent )
 {
 	QFormLayout * form = new QFormLayout(this);
+	form->setRowWrapPolicy(QFormLayout::WrapLongRows);
 
-	QString dev = ConfigManager::inst()->value( "audiosdl", "device" );
-	m_device = new QLineEdit( dev, this );
+	m_playbackDeviceComboBox = new QComboBox(this);
 
-	form->addRow(tr("Device"), m_device);
+	populatePlaybackDeviceComboBox();
+
+	form->addRow(tr("Playback device"), m_playbackDeviceComboBox);
+
+	m_inputDeviceComboBox = new QComboBox(this);
+
+	populateInputDeviceComboBox();
+
+	form->addRow(tr("Input device"), m_inputDeviceComboBox);
 }
 
 
 
 
 void AudioSdl::setupWidget::saveSettings()
 {
-	ConfigManager::inst()->setValue( "audiosdl", "device",
-							m_device->text() );
+	const auto currentPlaybackDevice = m_playbackDeviceComboBox->currentText();
+	if (currentPlaybackDevice == s_systemDefaultDevice)
+	{
+		// Represent the default input device with an empty string
```

Comment:
I think this comment should refer to the "default playback device".

Reply:
Fixed with commit https://github.com/LMMS/lmms/pull/7421/commits/f78e147d702253313fbce71183beb533f8aa472c.